src/prognostic_equations/implicit/implicit_solver.jl

```diff
@@ -1008,6 +1023,8 @@ function update_implicit_equation_jacobian!(A, Y, p, dtγ)
             # entrainment and detrainment
             (; ᶜentrʲs, ᶜdetrʲs, ᶜturb_entrʲs) = p
             # This assumes entrainment and detrainment rates are constant in the Jacobian
+            ∂ᶜq_totʲ_err_∂ᶜq_totʲ =
+                matrix[@name(c.sgsʲs.:(1).q_tot), @name(c.sgsʲs.:(1).q_tot)]
```

Comment:
```suggestion
```
I don't think you need this, but you need the other entrainment terms for the extra tracers. You don't need to do it for this PR though.

Reply:
I added entrainment/detrainment. But I agree, lets do the implicit stuff in the next PR